### PR TITLE
Improve small issues

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by okuta** [comment]
Jenkins, test this please

File: cupy/core/_kernel.pyx [hunk 1]
```diff
@@ -379,16 +379,18 @@ cdef list _get_out_args_with_params(
...
         for p in out_params:
             if p.raw and not is_size_specified:
                 raise ValueError('Output array size is Undecided')
         return [ndarray(out_shape, t) for t in out_types]
 
-    shape = out_shape
-    for i in range(len(out_params)):
+    shape.reserve(len(out_shape))
+    for x in out_shape:
+        shape.push_back(x)
```
Comment (toslunar) [inline cupy/core/_kernel.pyx]: Was `shape = out_shape` slow?
```cxx
  /* "cupy/core/_kernel.pyx":388
 *         return [ndarray(out_shape, t) for t in out_types]
 *
 *     shape = out_shape             # <<<<<<<<<<<<<<
 *     for i in range(len(out_params)):
 *         a = out_args[i]
 */
  __pyx_t_8 = __pyx_convert_vector_from_py_Py_ssize_t(__pyx_v_out_shape); if (unlikely(PyErr_Occurred())) __PYX_ERR(1, 388, __pyx_L1_error)
  __pyx_v_shape = __pyx_t_8;
```

Comment (okuta) [reply cupy/core/_kernel.pyx]: Yes, it is slow.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failure: Reported the test flakiness (chainer/chainer#8481).

**Pull request merged**